### ISSUE TITLE
Fastlane gym does not use provisioning profile set by match  for adhoc

### ISSUE THREAD

**Issue by Vadimkomis** [fastlane/fastlane#16520]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Essentially on CircleCI I cannot generate my archive because it's using the production target and not the adhoc one, even though it's setup correctly. 
Locally it works as expected.
In addition, I also checked that I setup the keychain correctly via SSH into the ci

I've also looked at similar issues reported here:
https://github.com/fastlane/fastlane/issues/12849
https://github.com/fastlane/fastlane/issues/10745
https://github.com/fastlane/fastlane/issues/4694
https://github.com/fastlane/fastlane/issues/10458

##### Command executed

```
error: No profiles for 'com.xxx.mobile.ios.NotificationService' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'com.xxx.mobile.ios.NotificationService'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild. (in target 'NotificationService' from project 'xxx')
```
##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Environment

<details>
  <pre> 
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.15.5                                     |
| Ruby                        | 2.6.3                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.24.1 (Apple Git-126)          |
| Installation Source         | ~/.rbenv/versions/2.6.3/bin/fastlane        |
| Host                        | Mac OS X 10.15.5 (19F83c)                   |
| Ruby Lib Dir                | ~/.rbenv/versions/2.6.3/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2s  28 May 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.4                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:ios)
before_all do
  if is_ci?
    setup_circle_ci
  end
end

platform :ios do
  if is_ci
    password = SecureRandom.base64
    keychain_name = "fastlane"
    ENV["MATCH_KEYCHAIN_NAME"] = keychain_name
    ENV["MATCH_KEYCHAIN_PASSWORD"] = password

    create_keychain(
      name: keychain_name,
      default_keychain: true,
      unlock: true,
      timeout: 3600,
      lock_when_sleeps: true,
      password: password
    )
  end
	
  desc "Update iOS UDID's on the developer portal"
  lane :sync_device_info do
    register_devices(
      devices_file: "fastlane/Devicefile"
    )
  end

  desc "Sync team Code_signing assets"
  lane :sync_signing_assets do |options|
    sync_device_info
    selectedType = options[:type]
    match(
      app_identifier: ["com.xxx.mobile.iosadhoc","com.xxx.mobile.iosadhoc.NotificationServiceDev"],
      type: selectedType)
  end

  desc "Build for App Store submission for dev"
  lane :build_appstore do
    sync_signing_assets(type: "adhoc") 
    gym(
      output_directory: "build_appstore",
      export_method: "ad-hoc",
      export_options: {
        signingStyle: "manual",
        provisioningProfiles: {
          "com.xxx.mobile.iosadhoc": "match AdHoc com.xxx.mobile.xxx",
          "com.xxx.mobile.xxx.NotificationServiceDev": "match AdHoc com.xxx.mobile.xxx.NotificationServiceDev"
        }
      }

    )
  end

  lane :distribute_to_appstore do
    build_appstore
    pilot(
      skip_waiting_for_build_processing: true,
      ipa:"./build_appstore/Xxxx Dev.ipa"
    )
  end

  lane :distribute_to_firebase do
    build_appstore
    firebase_app_distribution(
        app: "xxxxx:xxxxxx",
        groups: "QAs",
        testers: "xxx@xxx.com",
        release_notes: "Lots of amazing new features to test out!",
        ipa_path:"./build_appstore/Xxxx Dev.ipa",
        firebase_cli_path: "/usr/local/bin/firebase"
    )
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
apple_id "xxx@xxx.com"
itc_team_name "xxx"
itc_team_id "xxx"

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.148.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                    | Version | Update-Status |
| ----------------------------------------- | ------- | ------------- |
| fastlane-plugin-versioning                | 0.4.3   | ✅ Up-To-Date  |
| fastlane-plugin-firebase_app_distribution | 0.1.4   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                       | Version      |
| ----------------------------------------- | ------------ |
| did_you_mean                              | 1.3.1        |
| slack-notifier                            | 2.3.2        |
| atomos                                    | 0.1.3        |
| CFPropertyList                            | 3.0.2        |
| claide                                    | 1.0.3        |
| colored2                                  | 3.1.2        |
| nanaimo                                   | 0.2.6        |
| xcodeproj                                 | 1.16.0       |
| rouge                                     | 2.0.7        |
| xcpretty                                  | 0.3.0        |
| terminal-notifier                         | 2.0.0        |
| terminal-table                            | 1.8.0        |
| plist                                     | 3.5.0        |
| multipart-post                            | 2.0.0        |
| word_wrap                                 | 1.0.0        |
| public_suffix                             | 2.0.5        |
| tty-screen                                | 0.7.1        |
| tty-cursor                                | 0.7.1        |
| tty-spinner                               | 0.9.3        |
| babosa                                    | 1.0.3        |
| colored                                   | 1.2          |
| highline                                  | 1.7.10       |
| commander-fastlane                        | 4.4.6        |
| unf                                       | 0.1.4        |
| domain_name                               | 0.5.20190701 |
| http-cookie                               | 1.0.3        |
| faraday-cookie_jar                        | 0.0.6        |
| fastimage                                 | 2.1.7        |
| gh_inspector                              | 1.1.3        |
| mini_magick                               | 4.10.1       |
| multi_xml                                 | 0.6.0        |
| rubyzip                                   | 1.3.0        |
| security                                  | 0.1.3        |
| xcpretty-travis-formatter                 | 1.0.0        |
| naturally                                 | 2.2.0        |
| simctl                                    | 1.6.8        |
| jwt                                       | 2.1.0        |
| unicode-display_width                     | 1.7.0        |
| addressable                               | 2.7.0        |
| excon                                     | 0.73.0       |
| faraday                                   | 1.0.1        |
| unf_ext                                   | 0.0.7.7      |
| faraday_middleware                        | 1.0.0        |
| json                                      | 2.3.0        |
| dotenv                                    | 2.7.5        |
| bundler                                   | 2.1.4        |
| httpclient                                | 2.8.3        |
| multi_json                                | 1.14.1       |
| signet                                    | 0.14.0       |
| os                                        | 1.1.0        |
| memoist                                   | 0.16.2       |
| googleauth                                | 0.12.0       |
| mini_mime                                 | 1.0.2        |
| retriable                                 | 3.1.2        |
| declarative-option                        | 0.1.0        |
| declarative                               | 0.0.10       |
| uber                                      | 0.1.0        |
| representable                             | 3.0.4        |
| google-api-client                         | 0.38.0       |
| digest-crc                                | 0.5.1        |
| google-cloud-errors                       | 1.0.0        |
| google-cloud-env                          | 1.3.1        |
| google-cloud-core                         | 1.5.0        |
| google-cloud-storage                      | 1.26.1       |
| emoji_regex                               | 1.0.1        |
| aws-eventstream                           | 1.1.0        |
| aws-sigv4                                 | 1.1.3        |
| aws-partitions                            | 1.320.0      |
| jmespath                                  | 1.4.0        |
| aws-sdk-core                              | 3.96.1       |
| aws-sdk-kms                               | 1.31.0       |
| aws-sdk-s3                                | 1.66.0       |
| forwardable                               | 1.2.0        |
| logger                                    | 1.4.1        |
| stringio                                  | 0.0.2        |
| ipaddr                                    | 1.2.2        |
| openssl                                   | 2.1.2        |
| ostruct                                   | 0.1.0        |
| strscan                                   | 1.0.3        |
| date                                      | 2.0.0        |
| fileutils                                 | 1.3.0        |
| io-console                                | 0.4.8        |
| zlib                                      | 1.0.0        |
| mini_portile2                             | 2.4.0        |
| nokogiri                                  | 1.10.9       |
| rexml                                     | 3.2.4        |
| psych                                     | 3.1.0        |
| mutex_m                                   | 0.1.0        |
| oj                                        | 3.9.2        |
| bigdecimal                                | 1.4.4        |
| webrick                                   | 1.5.0        |
| fastlane-plugin-versioning                | 0.4.3        |
| fastlane-plugin-firebase_app_distribution | 0.1.4        |
</details>

*generated on:* **2020-05-26**
</details>

 </pre>
</details>

**Comment by Marnus29** [comment]
Hie, you need to make some changes in the structure. Try it once

**Comment by Vadimkomis** [comment]
@Marnus29 what do you mean? Can you give an example ?

**Comment by pattiriggs** [comment]
I have same question and didn't get satisfied solution.

**Comment by janziemba** [comment]
+1
I'm experiencing the same issue.

**Comment by Vadimkomis** [comment]
It wold be great to hear from someone on the team.

**Comment by Vadimkomis** [comment]
I finally was able to solve this issue. My solution was not the most elegant, but it works now on CircleCI. What I ended up doing was to remove the NotificationService target and then added a new one instead. That fixed the issue for me.
I think it had something to do with naming, but it's not something that I was able to verify. In addition, why did it work locally and not on CI, is still a mystery to me.  I hope this helps to anyone that experience a similar issue to me.
